sklearn/linear_model/tests/test_sgd.py

```diff
@@ -2156,3 +2156,26 @@ def test_validation_mask_correctly_subsets(monkeypatch):
     X_val, y_val = mock.call_args[0][1:3]
     assert X_val.shape[0] == int(n_samples * validation_fraction)
     assert y_val.shape[0] == int(n_samples * validation_fraction)
+
+
+def test_sgd_error_on_zero_validation_weight():
+    # Test that SGDClassifier raises error when all the validation samples
+    # have zero sample_weight
+    X, Y = iris.data, iris.target
+    sample_weight = np.zeros_like(Y)
+    validation_fraction = 0.4
+
+    clf = linear_model.SGDClassifier(
+        early_stopping=True,
+        tol=1e-3,
+        max_iter=1000,
+        validation_fraction=validation_fraction,
+        random_state=0,
+    )
```

Comment:
To simplify the test, we can remove the parameters that are not needed to trigger the error:

```suggestion
    clf = linear_model.SGDClassifier(
        early_stopping=True, validation_fraction=validation_fraction, random_state=0
    )
```